R/checks.R

```diff
@@ -1007,4 +1007,55 @@ check_invalid_form1.1 <- function(form) {
         (ptstatus == 1 & is_invalid_na_or_n(condate)) |
         (ptstatus == 2 & is_invalid_or_na(condate))
     )
+}
+
+#' Check that all entries in form3.1 are filled with valid values
+#' 
+#' @param form dataframe containing form3.1
+#' @return a dataframe containing problematic entries
+#' 
+#' @import tidyverse
+#' @export
+check_invalid_form3.1_2 <- function(form) {
+  form %>% 
+    filter(
+      is_invalid_na_or_n(injdate) | injdate == 0 |
+        is_invalid_na_or_n(alcohol) | alcohol == 0 |
+        is_invalid_na_or_n(selfi) | selfi == 0 |
+        
+        is_invalid_na_or_n(trans) |
+        (trans == 1 &
+           (is_invalid_na_or_n(transsp) | transsp == 0 |
+              (transsp == 6 & (is_invalid_na_or_n(bhelm) | bhelm == 0)) |
```

Comment:
if transsp != 6, you still have to check that  bhelm is not invalid, NA, or 0

Reply:
I would suggest do something like this:
(trans == 1 & (is_invalid_na_or_n(transsp) | transsp == 0 | is_invalid_or_na(bhelm)  | bhem == 0 | .... )  |
(transsp == 6 & is_n(bhelm)) ...

Reply:
Also I would strongly suggest you putting these outside of filter.
Use a variable like is_transport_invalid <- .... to store whether transport column is valid.